### PR TITLE
EEP 54: Rename the `reason` key to `cause` and make it optional

### PR DESCRIPTION
**Pull request by bjorng**

### CODE REVIEW

File: eeps/eep-0054.md [hunk 1]
```diff
@@ -76,11 +76,10 @@ of two-tuples.  As an indication that extended error info is available,
 we propose adding an `{error_info,ErrorInfoMap}` tuple to `ExtraInfo`
 in the first element in the stacktrace.
 
-The map `ErrorInfoMap` contains further information about the error.
-This should contain at least the key `reason`.  The corresponding
-value could contain additional information about the error.  When there
-is no additional information avaiable, the value is a dummy value such
-as `none` or `[]`.
+The map `ErrorInfoMap` may contain further information about the error
+or hints how to handle the error.  Currently, only the optional key
```
Comment (paulo-ferraz-oliveira) [inline eeps/eep-0054.md:80]: hints > hints on
or
hints > hints about
?

File: eeps/eep-0054.md [hunk 2]
```diff
@@ -122,32 +121,29 @@ could return:
...
-`erlang:format_error/4` only examines the arguments for `element/2`.
-
-The value for the key `reason` term will typically have a meaningful
-value when an error occurs in a BIF that depends on the internal state
-in the runtime system (such as `register/2` or the ETS BIFs), or for
-BIFs with complex arguments (such as `system_flag/2`) that would
-make it tedious and error prone to figure out which argument was
-in error.
+modules.  Note that the `cause` key in the map is optional.  If the
+`cause` key *is* present, the particular value of for a particular
```
Comment (paulo-ferraz-oliveira) [inline eeps/eep-0054.md:125]: of for > of
or
of for > for
?

File: eeps/eep-0054.md [hunk 3]
```diff
@@ -206,10 +202,23 @@ be implemented:
...
 
-The `register/2` BIF will provide specific error reasons for two of
-the possible failure reasons.  If the reason is not one of the two,
-`format_error/4` will figure out the other reasons based on the
-arguments.
+Note that different strategies for different BIFS are used for
+determining the extended error information:
+
+* First the `system_limit` exception is handled (regardless of which
+  BIF that was called).  No extended error information is returned,
```
Comment (paulo-ferraz-oliveira) [inline eeps/eep-0054.md:209]: I believe `that` can be suppressed here.
or you could also write "regardless of the BIF that was called"

File: eeps/eep-0054.md [hunk 4]
```diff
@@ -371,14 +379,19 @@ analysis of the error reason, there is a much lower risk that error
...
-### Why are the the reasons in the `ErrorInfoMap` undocumented?
+### Why is the map key named `cause` and not `reason`?
+
+To avoid confusion with the exception reason.
+
+
+### Why is the value for `cause` in the `ErrorInfoMap` undocumented?
 
 The reason in the `ErrorInfoMap` is not meant to be used for
 programmatically figure out why an error occurred, but only to be used
```
Comment (paulo-ferraz-oliveira) [inline eeps/eep-0054.md:390]: figure > figuring
otherwise "to programmatically figure"

**Comment by bjorng** [comment]
@paulo-ferraz-oliveira Thanks for the proof-reading! Fixed.

**Pull request merged**